### ISSUE TITLE
K8s resources get stuck in "Pending" after successful deploy

### ISSUE THREAD

**Issue by milas** [tilt-dev/tilt#4557]
### Expected Behavior
* `KubernetesDiscovery` API spec should be updated to include UIDs for deployed services/deployments/pods to watch for Pods

### Current Behavior
* Looks like a race condition where sometimes it misses them; even though EngineState knows about them, the spec isn't being updated, so Pods are never discovered

### Steps to Reproduce
(Race condition - appears to happen intermittently on K8s projects)

### Context
Almost assuredly a regression in v0.20.3 from #4528 or other recent Pod watch API migration PR

**Comment by nicks** [comment]
Here's the repro case that I've been using, which I can get to repro it frequently:

```
include('../tilt-example-go/3-recommended/Tiltfile')
load('ext://namespace', 'namespace_create')
namespace_create('argo')
namespace_create('argo-events')

for i in range(20):
  name = 'example-%d' % i
  k8s_yaml(blob('''apiVersion: apps/v1
kind: Deployment
metadata:
  name: %s
  %s
  labels:
    app: %s
spec:
  selector:
    matchLabels:
      app: %s
  template:
    metadata:
      labels:
        app: %s
    spec:
      containers:
      - name: %s
        image: busybox
        command: ["sh", "-c", "echo \\"Serving files on port 8000: $POD_NAME\\" && busybox httpd -f -p 8000"]
        env:
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
''' % (name, name, name, name, name)))
```

where tilt-example-go is in a sibling directory.

My current hypothesis is that the necessary conditions are:
1) Some resources in uncategorized that get deployed first (the namespaces)
2) Some local_resources that get run next (this is what tilt-example-go provides)
3) A boatload of services being deployed

which seems to trigger the conditions necessary for the bug. Milas and I have narrowed it down a bit and have some ideas on what could be happening.

**Comment by omh1280** [comment]
Thanks for the hard work, folks!

**Comment by milas** [comment]
This should be fixed in [v0.20.4+](https://github.com/tilt-dev/tilt/releases/latest). If you continue to see issues, please feel free to comment here or open a new issue.
